react-native/react/login2/register/set-public-name/index.js

```diff
@@ -15,11 +15,14 @@ export default class SetPublicName extends Component {
     }
   }
 
-  submit () {
-    this.props.submit(this.state.deviceName)
+  onSubmit () {
+    this.props.onSubmit(this.state.deviceName)
   }
 
   render () {
+    const dupeName = this.props.existingDevices && this.props.existingDevices.indexOf(this.state.deviceName) !== -1
```

Comment:
There should be a todo here to do more dupe checking, because don't we strip certain characters and trim the space? so you could "submit" a duplicate name. (although I think it would fail on the go side)